spec/consensus/light-client/detection.md

```diff
@@ -1,3 +1,459 @@
+		   - possibly all the other pairs *(f,h)* from *State* from full
+		     nodes *f* that where used to find the fork (the primary,
+		     all involved secondaries)
+	- It submits this evidence
+	- It flags the light client to stop
+- Expected Postcondition
+    - It "terminates everything". TODO: should this be described in a nicer
+  control flow? How should this be escalated to the whole light client?
+
+#### From the verifier
+
+```go
+VerifyHeaderAtHeight (untrustedHeight int64,
+                          trustedState TrustedState,
+			              addr Address) (TrustedState, error)
+```
+- Implementation remark
+  - signature deviates from current verification spec, which is
+    written with having bisection with the primary in mind. However,
+    we also need bisection with secondaries, so that we added the
+    Address `addr` of the full node the light client should do
+    bisection with. Also we changed the return values slightly, as we are
+    concurrently working on a more verification-oriented (verification
+    as in "model checking") light client verification spec
+    (verification 
+	as in [[verification]]).
+  - *startTime* and *endTime* are the local system time right after
+  invocation of `VerifyHeaderAtHeight` and right before the function returns, respectively.
+- Expected precondition
+  - The field `Time` of the signed header of `trustedState` is within *trustingPeriod* from *startTime*
+- Expected postcondition: 
+    - Returns `(trustedState, OK)` under [**[FN-LuckyCase]**](FN-LuckyCase-link), 
+  if the signed header of `trustedState`:
+    - is the header at height `untrustedHeight` of the blockchain, and 
+    - was generated within *trustingPeriod* from *endTime*
+    - corresponds to`return (trustedState,
+      nil)` in current verification spec.
+- Returns `(trustedState, EXPIRED)` under [**[FN-LuckyCase]**](FN-LuckyCase-link), if
+  the signed header of `trustedState`:
+    - is the header at height `untrustedHeight` of the blockchain, and 
+    - was generated after *endTime - trustingPeriod* 
+	- corresponds to `return (trustedState,
+      ErrHeaderNotWithinTrustedPeriod)` in current verification spec.
+- Error conditions
+  - precondition violated 
+  - [**[FN-LuckyCase]**](FN-LuckyCase-link) does not hold
+  - [**[FN-ManifestFaulty]**](FN-ManifestFaulty-link) holds
+
+## Solution
+
+Shared data of the light client
+- a pool of full nodes *FullNodes* that have not been contacted before
+- peer set called *Secondaries*
+- primary
+- State
+
+
+
+The problem is solved by calling  the function `FailureDetector` with
+a header *hd* that has
+just been verified by the verifier as a parameter. *trustedState*
+should be "a possibly old"
+trusted state to increase the likelihood of detecting a fork.
+
+
+```go
+func FailureDetector(hd Header, trustedState TrustedState)  {
+	for i, s range Secondaries {
+		sh := Commit(s,hd.height)
+		if validateSignedHeaderAndVals(sh,...) fails {
```

Comment:
What was `validateSignedHeaderAndVals` ?

Reply:
That is a function defined in [[verification]]. I have to add a reference. Thanks!